src/overlays/actors/ovl_En_Toryo/z_en_toryo.c

```diff
@@ -29,27 +30,337 @@ const ActorInit En_Toryo_InitVars = {
+            this->unk_1E4 = 5;
+        } else {
+            func_8002F434(&this->actor, globalCtx, 0x22, 100.0f, 10.0f);
+        }
+        return;
+    }
+
+    if (this->unk_1E4 == 0) {
+        if (func_8002F194(&this->actor, globalCtx) != 0) {
+            this->unk_1E0 = func_8002F368(globalCtx);
+            if (this->unk_1E0 != 0) {
+                player->actor.textId = func_80B20634(this, globalCtx);
+                this->actor.textId = player->actor.textId;
+            }
+            this->unk_1E4 = 1;
+            return;
+        }
+
+        func_8002F374(globalCtx, &this->actor, &sp32, &sp30);
+        if ((sp32 >= 0) && (sp32 < 0x141) && (sp30 >= 0) && (sp30 < 0xF1)) {
+            this->actor.textId = func_80B206A0(this, globalCtx);
+            func_8002F298(&this->actor, globalCtx, 100.0f, 10);
+        }
+    }
+}
+
+void func_80B20914(EnToryo* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80B20768(this, globalCtx);
+    if (this->unk_1E4 != 0) {
+        this->stateFlags |= 0x10;
+    } else {
+        this->stateFlags &= ~0x10;
+    }
+}
+
+void EnToryo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnToryo* this = THIS;
+    ColliderCylinder* collider = &this->collider;
+    Player* player = PLAYER;
+    f32 rot;
+
+    Collider_CylinderUpdate(thisx, collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, (Collider*)collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B203D8.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B205CC.s")
+    if ((this->stateFlags & 0x8) != 0) {
+        this->unk_1EC.unk_18.x = player->actor.posRot2.pos.x;
+        this->unk_1EC.unk_18.y = player->actor.posRot2.pos.y;
+        this->unk_1EC.unk_18.z = player->actor.posRot2.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20634.s")
+        if ((this->stateFlags & 0x10) != 0) {
+            func_80034A14(thisx, &this->unk_1EC, 0, 4);
+            return;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B206A0.s")
+        rot = thisx->yawTowardsLink - thisx->shape.rot.y;
+        if ((rot < 14563.0f) && (rot > -14563.0f)) {
+            func_80034A14(thisx, &this->unk_1EC, 0, 2);
+        } else {
+            func_80034A14(thisx, &this->unk_1EC, 0, 1);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20768.s")
+void EnToryo_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnToryo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20914.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                          EnToryo_overrideLimbDraw, EnToryo_postLimbDraw, thisx);
```

Comment:
you can pass `this` instead of `thisx` to the skelanime functions, if it matches